server/controllers/BasicServices.js

```diff
@@ -1,164 +1,320 @@
+      let responseHeader = await RestResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+      let sentResp = RestResponseBuilder.buildResponse(res, undefined, responseBody, responseHeader);
+      responseCode = sentResp.code;
+      responseBodyToDocument = sentResp.body;
     });
+  ExecutionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, responseCode, req.body, responseBodyToDocument);
 };
 
-module.exports.informAboutReleaseHistoryInGenericRepresentation = function informAboutReleaseHistoryInGenericRepresentation (req, res, next, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.informAboutReleaseHistoryInGenericRepresentation(user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.informAboutReleaseHistoryInGenericRepresentation = async function informAboutReleaseHistoryInGenericRepresentation (req, res, next, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.OK;
+  let responseBodyToDocument = {};
+  await BasicServices.informAboutReleaseHistoryInGenericRepresentation(req.url)
+    .then(async function (responseBody) {
+      responseBodyToDocument = responseBody;
+      let responseHeader = await RestResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+      RestResponseBuilder.buildResponse(res, responseCode, responseBody, responseHeader);
     })
-    .catch(function (response) {
-      utils.writeJson(res, response);
+    .catch(async function (responseBody) {
+      let responseHeader = await RestResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+      let sentResp = RestResponseBuilder.buildResponse(res, undefined, responseBody, responseHeader);
+      responseCode = sentResp.code;
+      responseBodyToDocument = sentResp.body;
     });
+  ExecutionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, responseCode, req.body, responseBodyToDocument);
 };
 
-module.exports.inquireOamRequestApprovals = function inquireOamRequestApprovals (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.inquireOamRequestApprovals(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
+module.exports.inquireOamRequestApprovals = async function inquireOamRequestApprovals (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+  let responseCode = responseCodeEnum.code.NO_CONTENT;
+  let responseBodyToDocument = {};
+  await BasicServices.inquireOamRequestApprovals(body, user, xCorrelator, traceIndicator, customerJourney, req.url, NEW_RELEASE_FORWARDING_NAME)
```

Comment:
undefined variable "**NEW_RELEASE_FORWARDING_NAME**" causing the following error and terminating the application , 

```

ReferenceError: NEW_RELEASE_FORWARDING_NAME is not defined
    at SwaggerRouter.redirectServiceRequestInformation (..\AirInterfacePowerSaver\server\controllers\BasicServices.js:185:124)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.router.js:99:40
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
    at ..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:346:12)
    at next (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:280:10)
    at ..\AirInterfacePowerSaver\server\node_modules\openbackhaul-oas3-tools\dist\middleware\swagger.parameters.js:52:13
    at Layer.handle [as handle_request] (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (..\AirInterfacePowerSaver\server\node_modules\express\lib\router\index.js:328:13)
```

Reply:
Done!